content/staging.go

```diff
@@ -394,25 +354,58 @@ func (cm *ContentManager) CreateAggregate(ctx context.Context, conts []util.Cont
 }
 
 // getStagedContentsGroupedByLocation gets the active(pin) contents only for aggregation and consolidation
-func (cm *ContentManager) getStagedContentsGroupedByLocation(ctx context.Context, zoneContID uint) (map[string][]util.Content, error) {
-	var conts []util.Content
+func (cm *ContentManager) getStagedContentsGroupedByLocation(ctx context.Context, zoneContID uint) (string, []util.Content, int64, error) {
+	var dstLocation string
+	var curMax int64
+
+	dataByLoc := make(map[string]int64)
 	var contsBatch []util.Content
-	if err := cm.db.Where("active and aggregated_in = ?", zoneContID, 5, nil).Select("size", "location").FindInBatches(&contsBatch, 500, func(tx *gorm.DB, batch int) error {
-		conts = append(conts, contsBatch...)
+	if err := cm.db.Where("active and aggregated_in = ? and location <> ?", zoneContID, constants.ContentLocationLocal).Select("size", "location").FindInBatches(&contsBatch, 500, func(tx *gorm.DB, batch int) error {
+		for _, c := range contsBatch {
+			// temp: dont ever migrate content back to primary instance for aggregation, always prefer elsewhere
+			if c.Location == constants.ContentLocationLocal {
+				continue
+			}
+
+			dataByLoc[c.Location] = dataByLoc[c.Location] + c.Size
+
+			canAddContent, err := cm.shuttleMgr.CanAddContent(c.Location)
+			if err != nil {
+				return err
+			}
+
+			if dataByLoc[c.Location] > curMax && canAddContent {
+				curMax = dataByLoc[c.Location]
+				dstLocation = c.Location
+			}
+		}
 		return nil
 	}).Error; err != nil {
-		return nil, err
+		return "", nil, 0, err
 	}
 
-	out := make(map[string][]util.Content)
-	for _, c := range conts {
-		out[c.Location] = append(out[c.Location], c)
+	if dstLocation == "" {
+		return "", nil, 0, fmt.Errorf("zone: %d failed to consolidate as no destination location could be determined", zoneContID)
 	}
 
-	if len(out) == 0 {
-		return nil, fmt.Errorf("no location for staged contents")
+	var toMove []util.Content
+	var toMoveBatch []util.Content
+	if err := cm.db.Where("active and aggregated_in = ? and location <> ?", zoneContID, constants.ContentLocationLocal).FindInBatches(&toMoveBatch, 500, func(tx *gorm.DB, batch int) error {
```

Comment:
i dont think we want the location <> ? part here. we still need to move local contents to the shuttle that is going to be aggregating and making the deal

Reply:
ah right, not from shuttle to local, but local to shuttle